crates/ide_assists/src/utils/gen_trait_fn_body.rs

```diff
@@ -326,3 +326,180 @@ fn gen_hash_impl(adt: &ast::Adt, func: &ast::Fn) -> Option<()> {
     ted::replace(func.body()?.syntax(), body.clone_for_update().syntax());
     Some(())
 }
+
+/// Generate a `PartialEq` impl based on the fields and members of the target type.
+fn gen_partial_eq(adt: &ast::Adt, func: &ast::Fn) -> Option<()> {
+    fn gen_discriminant() -> ast::Expr {
+        let root = make::ext::ident_path("core");
+        let submodule = make::ext::ident_path("mem");
+        let fn_name = make::ext::ident_path("discriminant");
+        let fn_name = make::path_concat(submodule, fn_name);
+        make::expr_path(make::path_concat(root, fn_name))
```

Comment:
It would probably be useful to add a convenience helper for building paths from `&[&str]`.

Reply:
Ah, good call. Just learned we can do `make::path_from_text` instead, which I'll replace all these calls with!